node/src/types/block.rs

```diff
@@ -825,6 +973,22 @@ mod tests {
         assert_eq!(finalized_block, decoded);
     }
 
+    #[test]
+    fn bytesrepr_roundtrip() {
+        let mut rng = TestRng::new();
+        let block = Block::random(&mut rng);
+        bytesrepr::test_serialization_roundtrip(&block);
+    }
+
+    #[test]
+    fn bytesrepr_roundtrip_era_end() {
+        let mut rng = TestRng::new();
+        let finalized_block = FinalizedBlock::random(&mut rng);
+        if let Some(era_end) = finalized_block.era_end() {
```

Comment:
Let's loop, creating random finalized blocks until we get `Some(era_end)`.  That way we're at least guaranteed on every test run to be actually testing this roundtrip.

Reply:
Aaah, I see your point